### DISCUSSION TITLE
windows下用upx尝试压缩linux mipsel版本给路由器会导致无法运行程序

### DISCUSSION THREAD

**Discussion by wsvg14** [IrineSistiana/mosdns#36] [Q&A] [answered]
windows 10 最新20xx版本下用upx for windows压缩mipsel版本给路由器
从0.16一直到0.23都试过，会导致程序无法正常运行，可知不是版本问题
运行自动脚本在ps能看到进程但是用netstat无法看到端口被绑定，终端手动运行主程序会直接不输出任何东西只能强制退出
之后更新上来0.24，不用upx处理，直接可运行
不知道是windows版本upx对于linux执行文件压缩的兼容性不好还是别的什么问题

其实对于路由器来说mosdns这个大小确实很不好塞，对于很多常见的16mb emmc路由器来说，原始文件高达13mb的情况下基本上得上u盘搞opt，非常不方便整合（虽然对于我给宿舍弄的128m emmc路由器来说硬塞原始文件没有任何问题。。。），还是希望有办法安全的压下来容量

**Comment by IrineSistiana** [comment, marked as answer]
upx 的老问题 。你回退 upx 版本试试？
https://github.com/IrineSistiana/mos-chinadns/issues/34

**Reply by wsvg14** [reply]
已经测试最新0.25.1版本使用3.95版本upx可以正常运行，确实是最新3.96版本upx对于文件的处理问题
建议以后加个提示尽量使用3.95版本upx，避免出现这种诡异的问题（点名批评upx又在整活）
